### DISCUSSION TITLE
Firebase cannot be imported

### DISCUSSION THREAD

**Discussion by FreakFrog** [quasarframework/quasar#9163] [General - Components / Directives / etc]
the error :Module build failed (from ./node_modules/@quasar/app/lib/webpack/loader.transform-quasar-imports.js):
  Error: Unknown import from Quasar: firebase

/plugins/firebase.js

import firebase from "firebase/app";

import "firebase/auth";
import "firebase/firestore";

const firebaseConfig = {
  apiKey: "**************************",
  authDomain: "test-bb237.firebaseapp.com",
  projectId: "test-bb237",
  storageBucket: "test-bb237.appspot.com",
  messagingSenderId: "459314904928",
  appId: "1:459314904928:web:2167a4e676a779a6a0a805",
  measurementId: "G-V394PBZTM0"
};
const fireApp = firebase.initializeApp(firebaseConfig);

export const AUTH = fireApp.auth();
export const fireStore = firebase.firestore();
export const timeStamp = firebase.firestore.FieldValue.serverTimestamp;
//////////////////////
quasarconf     // plugins['firebase']

*note* i installed firebase and double checked
is there something I'm missing?

**Comment by ontwikkelfabriek** [comment]
I do not know why you get the error message. But I do think you are confusing Quasar `plugins` with Quasar `boot files`.  So instead of initializing Firebase in `/plugins/firebase.js` you probably want to that in the boot dir `/boot/firebase.js` . Additionally in quasarconf use: boot['firebase']`.

Boot files:
https://quasar.dev/quasar-cli/boot-files

An example of a Quasar plugin:
https://quasar.dev/quasar-plugins/app-fullscreen

**Reply by FreakFrog** [reply]
the guides i found online did firebase as a plugin, I will try to use it as a boot then.

**Reply by metalsadman** [reply]
> the guides i found online did firebase as a plugin, I will try to use it as a boot then.

you must be lookin at quasar pre version 1 articles, in quasar version 1 and up "app plugins" `plugin` was changed in place of `boot` files. https://quasar.dev/start/upgrade-guide#initial-steps #9.
